llmfoundry/models/layers/norm.py

```diff
@@ -111,9 +112,65 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
                             self.eps).to(dtype=x.dtype)
 
 
+class TritonRMSNorm(torch.nn.Module):
+
+    def __init__(
+        self,
+        normalized_shape: Union[int, List[int], torch.Size],
+        eps: float = 1e-5,
+        dropout_p: float = 0.0,
+        dtype: Optional[torch.dtype] = None,
+        device: Optional[torch.device] = None,
+    ):
+        if isinstance(normalized_shape, int):
+            hidden_size = normalized_shape
+        elif isinstance(normalized_shape, list):
+            hidden_size = math.prod(normalized_shape)
+        elif isinstance(normalized_shape, torch.Size):
+            hidden_size = math.prod(normalized_shape)
+
+        factory_kwargs = {'device': device, 'dtype': dtype}
+        super().__init__()
+        self.eps = eps
+        if dropout_p > 0.0:
+            self.drop = torch.nn.Dropout(dropout_p)
+        else:
+            self.drop = None
+        self.weight = torch.nn.Parameter(
+            torch.empty(hidden_size, **factory_kwargs))
+        self.register_parameter('bias', None)
+        torch.nn.init.ones_(self.weight)
+
+        try:
+            from flash_attn.ops.triton.layer_norm import rms_norm_fn
+            self.rms_norm_fn = rms_norm_fn
+        except ImportError:
+            raise ImportError(
+                'triton_rms_norm requires Flash Attention to be installed. ' +
+                'Please `pip install llm-foundry[gpu-flash2]`.')
```

Comment:
this should just say to install fa, because the model code can end up isolated from the foundry code

Reply:
👍